docs/README.md

```diff
@@ -218,6 +218,16 @@ tests are using at the link below:
 
 + [https://hub.docker.com/r/p4lang/behavioral-model/builds](https://hub.docker.com/r/p4lang/behavioral-model/builds)
 
+### Adding new test data
+
+To add a new input test with a sample P4 code file (under `testdata/p4_16_samples/` for example), one needs to:
+
+* Add the `*.p4` file to the `testdata/p4_16_samples/` directory. The file name might determine which test suite this test belongs to. Those are determined by cmake commands.
+  * For example, [any P4 file under `testdata/p4_16_samples/`](https://github.com/p4lang/p4c/blob/de2eaa085152abd0690660fbe561eaf5db7b2bf7/backends/p4test/CMakeLists.txt#L63) belongs to the [`p4` test suite](https://github.com/p4lang/p4c/blob/de2eaa085152abd0690660fbe561eaf5db7b2bf7/backends/p4test/CMakeLists.txt#L113) (meaning it will run with `make check-p4`).
+  * [File ending with `*-bmv2.p4`](https://github.com/p4lang/p4c/blob/de2eaa085152abd0690660fbe561eaf5db7b2bf7/backends/bmv2/CMakeLists.txt#L127) also belongs to the [`bmv2` test suite](https://github.com/p4lang/p4c/blob/de2eaa085152abd0690660fbe561eaf5db7b2bf7/backends/bmv2/CMakeLists.txt#L200) (meaning it will run with `make check-bmv2`).
+* Then generate expected outputs by `P4TEST_REPLACE=True make check` (or a more specific test command to save time).
```

Comment:
There is also the `-f` flag for `run-bmv2-test.py` and `run-p4-sample.py`.

Reply:
Got it! I didn't know them before. Let me try them out and include them in the doc.

Reply:
P4TEST_REPLACE=TRUE is basically equivalent, but sometimes you have an environment where that command is awkward to use.

Reply:
I have updated the doc.